pkg/cluster/storageaccounts.go

```diff
@@ -98,14 +100,63 @@ func (m *manager) populateRegistryStorageAccountName(ctx context.Context) error
 		return nil
 	}
 
-	rc, err := m.imageregistrycli.ImageregistryV1().Configs().Get(ctx, "cluster", metav1.GetOptions{})
+	registryConfig, err := m.getRegistryConfig(ctx)
 	if err != nil {
 		return err
 	}
 
-	m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
-		doc.OpenShiftCluster.Properties.ImageRegistryStorageAccountName = rc.Spec.Storage.Azure.AccountName
-		return nil
-	})
+	accountNameMutator := getAccountNameMutator(ctx, registryConfig)
+
+	m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, accountNameMutator)
 	return err
 }
+
+func (m *manager) getRegistryConfig(ctx context.Context) (*v1.Config, error) {
+	configName := "cluster"
+
+	registryConfig, err := m.imageregistrycli.ImageregistryV1().Configs().Get(ctx, configName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return registryConfig, nil
+}
```

Comment:
This is superfluous imo. If you still want to have it in a function, which i wouldn't do , you can simplify by having 
```go 
func (m *manager) getRegistryConfig(ctx context.Context) (*v1.Config, error) {
	configName := "cluster"

	return m.imageregistrycli.ImageregistryV1().Configs().Get(ctx, configName, metav1.GetOptions{})
}
```
or even 

```go 
func (m *manager) getRegistryConfig(ctx context.Context) (*v1.Config, error) {
    return m.imageregistrycli.ImageregistryV1().Configs().Get(ctx, "cluster", metav1.GetOptions{})
}
```
but this becomes really obvious here that you don't need the function

Reply:
Interesting point, but I added the function because:
1- I don't want to know how the registry is obtained, I prefer to delegate it to a function so I hide the details when I am reading "populateRegistryStorageAccountName" method. It also favors SRP principle: if the way the registry config is obtained changes, we just have to touch this function and not "populateRegistryStorageAccountName" (same if it is used in more places).
2- it also adds the explanatory variable "configName" because the second parameter to Get() was hardcoded and at first sight was not clear what "cluster" meant. Now it is clear that refers to the config name.
🙂

Reply:
But you turn one line of code, which already follows srp , because it is already its own function , into 6 , and it brings little to no value here.

Reply:
Strictly speaking, if we don't create the aux function to get the config registry, we are breaking the SRP principle:  if the way the registry is obtained changes, we will touch the "populateRegistryStorageAccountName" function when in reality, this function should not be responsible for obtaining the registry config, the same way it should not be responsible for creating the mutator function, "populateRegistryStorageAccountName" just delegates those responsibilities to others. 

I agree that in this case as we just have one line it is not that clear, but if we forgot about the number of lines and just focus on responsibility, I would keep the aux function.

One function should do one thing and do it right. 

A great [conference](https://www.youtube.com/watch?v=QvrNwfuWbCQ) about the topic by Uncle Bob.

🙂

Reply:
but the function is already created and is literally called GET, so SRP is followed. the only thing that you are doing is renaming it and making the code more complex. if we follow your idea we should also have a function for creating the variable and one for calling get with that variable. It makes no sense.

Reply:
> but the function is already created and is literally called GET, so SRP is followed. the only thing that you are doing is renaming it and making the code more complex. if we follow your idea we should also have a function for creating the variable and one for calling get with that variable. It makes no sense.

I don't think "we should also have a function for creating the variable and one for calling get with that variable". Is just about group responsibility together. As I mention, this is not 100% obvious here as it is just one line. 

If the way we get the config file in the future changes, we will need to change it X times where X is the number of times we get the config in our codebase. If you have it on a function, you just need to change it in one place, in the function definition.

I will remove the aux function here (as it is just one line) but I still think IMHO the reasoning makes sense. 🙂